### PR TITLE
Task #165665 chore: Integration with Bitly for making short Url

### PR DESCRIPTION
**Pull request by shindebalu**

### CODE REVIEW

File: src/com_tjnotifications/admin/config.xml [hunk 1]
```diff
@@ -24,6 +24,16 @@
 			label="COM_TJNOTIFICATIONS_SUBSCRIPTIONS_SELECT_PHONE_NUMBER_FIELD" description="COM_TJNOTIFICATIONS_SUBSCRIPTIONS_SELECT_PHONE_NUMBER_FIELD_DESC"
 		/>
 
+		<field name="enable_url_shorting" type="radio" class="btn-group btn-group-yesno" default="0" label="COM_TJNOTIFICATIONS_SETTINGS_URL_SHORTING" description="COM_TJNOTIFICATIONS_SETTINGS_URL_SHORTING_DESC">
```
Comment (praneettekdi) [inline src/com_tjnotifications/admin/config.xml:27]: enable_url_shortening

File: src/com_tjnotifications/admin/config.xml [hunk 2]
```diff
@@ -24,6 +24,16 @@
...
+			<option value="0">JNO</option>
+			<option value="1">JYES</option>
+		</field>
+
+		<field name="url_shorting_provider" type="sql" showon="enable_url_shorting:1"
```
Comment (praneettekdi) [inline src/com_tjnotifications/admin/config.xml:32]: url_shortening_provider

File: src/com_tjnotifications/admin/config.xml [hunk 3]
```diff
@@ -24,6 +24,16 @@
...
+			default="" label="COM_TJNOTIFICATIONS_SETTINGS_CHOOSE_URL_SHORTENING_PLUGIN" description="COM_TJNOTIFICATIONS_SETTINGS_CHOOSE_URL_SHORTENING_PLUGIN_DESC" query="SELECT element, name FROM #__extensions WHERE type='plugin' AND folder='tjurlshortner' AND enabled=1" key_field="element" value_field="name" />
+
+		<field name="url_shorting_enabled_backends" type="tjnotificationsbackends" showon="enable_url_shorting:1" default="" multiple="multiple" label="COM_TJNOTIFICATIONS_SETTINGS_CHOOSE_BACKEND_SHORTENING" description="COM_TJNOTIFICATIONS_SETTINGS_CHOOSE_BACKEND_SHORTENING_DESC" />
```
Comment (praneettekdi) [inline src/com_tjnotifications/admin/config.xml:35]: replace all `shorting` with `shortening`

File: src/com_tjnotifications/admin/language/en-GB.com_tjnotifications.ini [hunk 4]
```diff
@@ -293,3 +293,9 @@ COM_TJNOTIFICATIONS_SUBSCRIPTIONS_PHONE_NUMBER_JOOMSOCIAL="JoomSocial"
 COM_TJNOTIFICATIONS_SUBSCRIPTIONS_SELECT_PHONE_NUMBER_FIELD="Select phone number field"
 COM_TJNOTIFICATIONS_SUBSCRIPTIONS_SELECT_PHONE_NUMBER_FIELD_DESC="Select Phone number field to get phone number"
 COM_TJNOTIFICATIONS_SUBSCRIPTIONS_SELECT_FIELD="-Select-"
+COM_TJNOTIFICATIONS_SETTINGS_URL_SHORTING="Enable URL shorting?"
+COM_TJNOTIFICATIONS_SETTINGS_URL_SHORTING_DESC="If set yes, URL from notification will short"
+COM_TJNOTIFICATIONS_SETTINGS_CHOOSE_URL_SHORTENING_PLUGIN="Choose URL shortening provider"
+COM_TJNOTIFICATIONS_SETTINGS_CHOOSE_URL_SHORTENING_PLUGIN_DESC="Choose URL shortening provider"
+COM_TJNOTIFICATIONS_SETTINGS_CHOOSE_BACKEND_SHORTENING="Choose backends for shortening URL"
+COM_TJNOTIFICATIONS_SETTINGS_CHOOSE_BACKEND_SHORTENING_DESC="Choose backends (like email,sms,whatsapp,push etc.) for shortening URL"
```
Comment (praneettekdi) [inline src/com_tjnotifications/admin/language/en-GB.com_tjnotifications.ini:301]: add space after each comma

Comment (praneettekdi) [reply src/com_tjnotifications/admin/language/en-GB.com_tjnotifications.ini:301]: What is backends?

Comment (praneettekdi) [reply src/com_tjnotifications/admin/language/en-GB.com_tjnotifications.ini:301]: `(like email,sms,whatsapp,push etc.)` => `(like Email, SMS, WhatsApp, Push etc.)`

File: src/com_tjnotifications/admin/models/fields/tjnotificationsbackends.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+	 * @since   2.0.0
+	 */
+	protected function getOptions()
+	{
+		$options  = array();
+		$backends = explode(',', TJNOTIFICATIONS_CONST_BACKENDS_ARRAY);
+
+		foreach ($backends as $keyBackend => $backend)
+		{
+			$options[] = JHtml::_('select.option', $backend,    Text::_('COM_TJNOTIFICATIONS_BACKEND_' . strtoupper($backend)));
```
Comment (praneettekdi) [inline src/com_tjnotifications/admin/models/fields/tjnotificationsbackends.php:46]: Use HTMLHELPER.

Remove extra space before `Text`

File: src/com_tjnotifications/admin/models/fields/tjnotificationsbackends.php [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+	 * @since   2.0.0
+	 */
+	protected function getOptions()
+	{
+		$options  = array();
+		$backends = explode(',', TJNOTIFICATIONS_CONST_BACKENDS_ARRAY);
+
+		foreach ($backends as $keyBackend => $backend)
+		{
+			$options[] = HTMLHelper::_('select.option', $backend,    Text::_('COM_TJNOTIFICATIONS_BACKEND_' . strtoupper($backend)));
```
Comment (praneettekdi) [inline src/com_tjnotifications/admin/models/fields/tjnotificationsbackends.php:47]: `$options[] = HTMLHelper::_('select.option', $backend,    Text::_('COM_TJNOTIFICATIONS_BACKEND_' . strtoupper($backend)));`

Change to 

`$options[] = HTMLHelper::_('select.option', $backend, Text::_('COM_TJNOTIFICATIONS_BACKEND_' . strtoupper($backend)));`

File: src/com_tjnotifications/admin/config.xml [hunk 7]
```diff
@@ -24,6 +24,16 @@
...
 
+		<field name="enable_url_shortening" type="radio" class="btn-group btn-group-yesno" default="0" label="COM_TJNOTIFICATIONS_SETTINGS_URL_SHORTING" description="COM_TJNOTIFICATIONS_SETTINGS_URL_SHORTING_DESC">
+			<option value="0">JNO</option>
+			<option value="1">JYES</option>
+		</field>
+
+		<field name="url_shortening_provider" type="sql" showon="enable_url_shortening:1"
+			default="" label="COM_TJNOTIFICATIONS_SETTINGS_CHOOSE_URL_SHORTENING_PLUGIN" description="COM_TJNOTIFICATIONS_SETTINGS_CHOOSE_URL_SHORTENING_PLUGIN_DESC" query="SELECT element, name FROM #__extensions WHERE type='plugin' AND folder='tjurlshortner' AND enabled=1" key_field="element" value_field="name" />
+
+		<field name="url_shortening_enabled_backends" type="tjnotificationsbackends" showon="enable_url_shortening:1" default="" multiple="multiple" label="COM_TJNOTIFICATIONS_SETTINGS_CHOOSE_BACKEND_SHORTENING" description="COM_TJNOTIFICATIONS_SETTINGS_CHOOSE_BACKEND_SHORTENING_DESC" />
```
Comment (manojLondhe) [inline src/com_tjnotifications/admin/config.xml:35]: set sms as default

**Review by manojLondhe — CHANGES_REQUESTED** [review]

**Pull request merged**